### ISSUE TITLE
Update all code to use common dialog functions for yes/no, ok/cancel, message box, etc. (Trac #71)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#75]
Currently, some code is using dialogs from pygtkhelpers.ui.dialogs while other code is using methods from the MainWindowController class.

Ideally customized dialog functions can be moved into the utility module.  One thing that needs to be determined is the effect that specifying the parent window has when creating a dialog.

Migrated from http://microfluidics.utoronto.ca/ticket/71
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "Currently, some code is using dialogs from pygtkhelpers.ui.dialogs while other code is using methods from the MainWindowController class.\n\nIdeally customized dialog functions can be moved into the utility module.  One thing that needs to be determined is the effect that specifying the parent window has when creating a dialog.\n\n", 
    "reporter": "cfobel", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "Update all code to use common dialog functions for yes/no, ok/cancel, message box, etc.", 
    "priority": "major", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2012-03-14T15:59:18", 
    "milestone": "Microdrop 1.0", 
    "owner": "cfobel", 
    "type": "defect"
}
```

**Comment by cfobel** [comment]
Trac update at 20120315T05:16:05: ryan changed status from "new" to "assigned"

**Comment by cfobel** [comment]
Trac update at 20120403T10:06:08: christian@fobel.net commented:

> commit d40dcbc9f03f181c77c57b3eb8398f69ade006e6
> Author: Christian Fobel <christian@fobel.net>
> Date:   Tue Apr 3 05:59:41 2012 -0400
> 
>     Migrate all get_text_input() calls to text_entr...
>     
>     Migrate all get_text_input() calls to text_entry_dialog()
>     See #75
>     
>     Note that this also commit also sets the default action for a text entry dialog
>     to OK.  This means that OK will be activated when ENTER is pressed inside the
>     text entry widget.

**Comment by cfobel** [comment]
Trac update at 20120403T10:06:13: christian@fobel.net commented:

> commit 88d8633359b22e039be5c9a19a9eaacafbfec95c
> Author: Christian Fobel <christian@fobel.net>
> Date:   Tue Apr 3 04:37:05 2012 -0400
> 
>     Move form_view_dialog to utility.gui (See #75)

**Comment by cfobel** [comment]
Trac update at 20120403T10:06:16: christian@fobel.net commented:

> commit 095fa63303bf1b14c4352a7ed6dcf0ee26753966
> Author: Christian Fobel <christian@fobel.net>
> Date:   Tue Apr 3 04:37:05 2012 -0400
> 
>     Generalize protocol_grid_dialog->form_view_dialog
>     
>     This will allow the dialog to be reused anywhere an input dialog is needed.
>     See #75.

**Comment by cfobel** [comment]
Trac update at 20120403T10:06:18: 
* christian@fobel.net commented:

> commit 3492c70d5e1107aa9933853b5a5c969f5677e57a
> Author: Christian Fobel <christian@fobel.net>
> Date:   Tue Apr 3 03:34:06 2012 -0400
> 
>     Update all code to use common dialog functions
>     
>     Update all code to use common dialog functions for yes/no, ok/cancel, message
>     box, etc.  These common dialogs belong to utility.gui
>     Fixes #75.


* christian@fobel.net changed resolution from "" to "fixed"
* christian@fobel.net changed status from "assigned" to "closed"

**Comment by cfobel** [comment]
Trac update at 20120403T10:06:46: christian@fobel.net commented:

> commit e07f721d4d962822e8c529222d97bde445928198
> Merge: 14c04f1918dc26b4311c5586eec5e035fefdaddd d40dcbc9f03f181c77c57b3eb8398f69ade006e6
> Author: Christian Fobel <christian@fobel.net>
> Date:   Tue Apr 3 06:03:08 2012 -0400
> 
>     Merge branch 'common_dialogs' (fixes #75)

**Comment by cfobel** [comment]
Trac update at 20120403T14:29:54: 
* ryan commented:

> I don't like the pygtkhelpers yesno dialog because it has the yes and no buttons in the wrong order. Windows conventions are to put buttons (from left to right) "OK/Cancel", "Yes/No". Is it possible to fix this or should we just replace it?
> 
> Also, the "default action" behaviour has a couple of problems. If the user changes the text, hitting enter no longer works because the focus is not on the text widget. Also if the user presses cancel, when a subsequent form dialog is opened, cancel is the new default.
> 
> What I'd ideally like to see is that when a user opens a form dialog, the text box should have the text selected so that the user can just start typing. Once they have replaced the text, they should be able to press enter to accept the change.


* ryan changed description from:

> Currently, some code is using dialogs from pygtkhelpers.ui.dialogs while other code is using methods from the MainWindowController class.
> 
> Ideally customized dialog functions can be moved into the utility module.  One thing that needs to be determined is the effect that specifying the parent window has when creating a dialog.

to:

> Currently, some code is using dialogs from pygtkhelpers.ui.dialogs while other code is using methods from the MainWindowController class.
> 
> Ideally customized dialog functions can be moved into the utility module.  One thing that needs to be determined is the effect that specifying the parent window has when creating a dialog.
> 
> 


* ryan changed resolution from "fixed" to ""
* ryan changed status from "closed" to "reopened"

**Comment by cfobel** [comment]
Trac update at 20120403T14:49:51: ryan commented:

> Actually, it seems that the "default action" part is working (I must have been testing a previous versions), but with the new version, the "glade" file needs to be added to the installer to make it work.

**Comment by cfobel** [comment]
Trac update at 20120403T22:15:45: 
* christian@fobel.net commented:

> commit 5e60b58cd9ffb76dff4b099426b0ac46417ec288
> Author: Christian Fobel <christian@fobel.net>
> Date:   Tue Apr 3 18:04:27 2012 -0400
> 
>     Customize yesno dialog in utility.gui for Windows
>     
>     Customize yesno dialog in utility.gui to use more consistent button ordering for
>     Windows.  Note that the Gnome Human Interface Guidelines place the affirmative
>     button at the far right.  This is backwards from what is typically found in
>     Windows.
>     See here for more details:
>         http://www.pygtk.org/docs/pygtk/class-gtkdialog.html#method-gtkdialog--set-alternative-button-order
>     
>     Note that these changes require uninstalling pygtkhelpers and reinstalling it using
>     install_dependencies.py
>     Fixes #75.


* christian@fobel.net changed resolution from "" to "fixed"
* christian@fobel.net changed status from "reopened" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "core" to "microdrop"
